libtenzir/src/partition_transformer.cpp

```diff
@@ -493,7 +457,16 @@ auto partition_transformer(
           data.indexer_chunks.emplace_back(qf.name(), chunk);
         }
       }
-      detail::shutdown_stream_stage(self->state.stage);
+      for (auto& [_, partition_data] : self->state.data) {
+        self->request(partition_data.builder, caf::infinite, atom::persist_v)
+          .then(
+            [](resource&) {
+              // nop
```

Comment:
This should deliver a response promise for the parent handler once they're all done, or at least have a TODO comment for that (with a follow-up story).